managed/src/main/java/com/yugabyte/yw/commissioner/tasks/subtasks/KubernetesCommandExecutor.java

```diff
@@ -705,4 +705,22 @@ private String generateHelmOverride() {
       throw new RuntimeException("Error writing Helm Override file!");
     }
   }
+
+  // Recursively traverses the override map and updates or adds the
+  // keys to source map.
+  private void mergeYaml(Map<String, Object> source, Map<String, Object> override) {
+    for (Entry<String, Object> entry : override.entrySet()) {
+      String key = entry.getKey();
+      if (source.containsKey(key)) {
```

Comment:
minor: i find it easier to read if we use negation to avoid depth. 

```
if (!source.contains(key) {
  ... continue
}
if (!override is map or source is map) {
  ... continue
}
mergeYaml recursive call
```

Reply:
Makes sense, I have updated it.